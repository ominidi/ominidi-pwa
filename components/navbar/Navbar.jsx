import React from 'react';
import PropTypes from 'prop-types';

const Navbar = ({ menuItems }) => (
  <nav className="navbar">
    <ul className="navbar__menu">
      {
        menuItems.map(item => (
          <li
            className="navbar__menu__item"
            key={item.href}
          >
            <a
              href={item.href}
              title={item.title}
              className="navbar__menu__link to-manifest"
            >
              {item.text}
            </a>
          </li>
        ))
      }
    </ul>
  </nav>
);

Navbar.propTypes = {
  menuItems: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string,
      title: PropTypes.string,
      text: PropTypes.string,
    }),
  ),
};

Navbar.defaultProps = {
  menuItems: [],
};

export default Navbar;
