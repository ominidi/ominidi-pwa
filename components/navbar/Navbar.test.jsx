import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Navbar />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render an menu item for each item present in the menuItem prop', () => {
    const menuItems = [
      {
        href: '/',
        title: 'Manifest',
        text: 'Manifest',
      },
      {
        href: '/photos',
        title: 'Photos',
        text: 'Photos',
      },
    ];

    const wrapper = shallow(<Navbar menuItems={menuItems} />);

    expect(wrapper.find('.navbar__menu__item').length).toEqual(menuItems.length);
    wrapper.find('.navbar__menu__link').forEach((item) => {
      expect(menuItems.some(element => element.href === item.prop('href'))).toBe(true);
      expect(menuItems.some(element => element.title === item.prop('title'))).toBe(true);
      expect(menuItems.some(element => element.text === item.text())).toBe(true);
    });
  });
});
