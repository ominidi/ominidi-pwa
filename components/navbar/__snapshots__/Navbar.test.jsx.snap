// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<nav
  className="navbar"
>
  <ul
    className="navbar__menu"
  />
</nav>
`;
